backend/hitas/oracle_migration/runner.py

```diff
@@ -494,101 +501,149 @@ def create_apartments(
-                select(apartment_market_price_indices).where(
-                    apartment_market_price_indices.c.max_price_index_id == market_price_index.id
-                )
-            ).fetchall()
+        mpi_improvements = get_latest_improvements(
+            market_price_indices, apartment_market_price_indices, apartment_oracle_id
+        )
+        for mpi_improvement in mpi_improvements:
+            new = ApartmentMarketPriceImprovement(
+                apartment=v,
+                name=mpi_improvement["name"],
+                completion_date=mpi_improvement["completion_date"],
+                value=mpi_improvement["value"],
+            )
 
-            for mpi_improvement in mpi_improvements:
-                new = ApartmentMarketPriceImprovement(
-                    apartment=v,
-                    name=mpi_improvement["name"],
-                    completion_date=mpi_improvement["completion_date"],
-                    value=mpi_improvement["value"],
-                )
-                bulk_mpi.append(new)
-                created[v.id][1].append(new)
+            # Check if improvement value should be moved to apartment.additional_work_during_construction instead
+            if mpi_improvement["excess"] == "000" and is_date_within_one_month(
+                new.completion_date, monthify(v.completion_date)
+            ):
+                mpi_awdc["improvements"].append(new)
+                mpi_awdc["date"] = mpi_improvement.calculation_date
+                continue
 
+            bulk_mpi.append(new)
+            count += 1
         if len(bulk_mpi) >= BULK_INSERT_THRESHOLD:
             ApartmentMarketPriceImprovement.objects.bulk_create(bulk_mpi)
             bulk_mpi = []
 
+        #
+        # Additional work during construction improvements
+        #
+
+        # One index improvements may contain the same improvements, but they are split into multiple separate
+        # improvements, so we need to compare the sums of both indices and if they match the sum value can be moved.
+        sum_cpi = sum(i.value for i in cpi_awdc["improvements"])
+        sum_mpi = sum(i.value for i in mpi_awdc["improvements"])
+        awdc = 0
+
+        if sum_cpi == sum_mpi or sum_cpi == 0 or sum_mpi == 0:
+            # CPI and MPI sums match, or one index is missing is completely.
+            awdc = max(sum_cpi, sum_mpi)
+        elif cpi_awdc["date"] != mpi_awdc["date"]:
+            # Mismatch on index improvement sums, use the latest calculation instead as its most up to date.
+            if cpi_awdc["date"] > mpi_awdc["date"]:
+                awdc = sum_cpi
+            else:
+                awdc = sum_mpi
+        else:
+            # Unable to convert improvements automatically.
+            # Print out and add them as normal improvements to be handled manually later.
+            bulk_cpi.extend(cpi_awdc["improvements"])
+            bulk_mpi.extend(mpi_awdc["improvements"])
+            print(
```

Comment:
Might want to add some title to the print like "Unhandled construction time improvement, adding as a regular improvement:". Also, would a multiline string look cleaner?

Reply:
Yep, changed this to multiline string :)